server/msteams/client.go

```diff
@@ -200,25 +200,40 @@ func NewManualClient(tenantID, clientID string, logService *pluginapi.LogService
 		client:     nil,
 	}
 
+	// Create an HTTP client with the middleware
+	httpClient := getHTTPClient()
+
 	cred, err := azidentity.NewDeviceCodeCredential(&azidentity.DeviceCodeCredentialOptions{
 		TenantID: tenantID,
 		ClientID: clientID,
 		UserPrompt: func(ctx context.Context, message azidentity.DeviceCodeMessage) error {
 			fmt.Println(message.Message)
 			return nil
 		},
+		ClientOptions: policy.ClientOptions{
+			Transport: getAuthClient(),
+		},
 	})
 	if err != nil {
 		fmt.Printf("Error creating credentials: %v\n", err)
 		return nil
 	}
 
-	client, err := msgraphsdk.NewGraphServiceClientWithCredentials(cred, teamsDefaultScopes)
+	authProvider, err := a.NewAzureIdentityAuthenticationProviderWithScopes(cred, append(teamsDefaultScopes, "offline_access"))
+	if err != nil {
+		fmt.Printf("Error creating credentials: %v\n", err)
+		return nil
+	}
+
+	adapter, err := msgraphsdk.NewGraphRequestAdapterWithParseNodeFactoryAndSerializationWriterFactoryAndHttpClient(authProvider, nil, nil, httpClient)
 	if err != nil {
-		fmt.Printf("Error creating client: %v\n", err)
+		fmt.Printf("Error creating credentials: %v\n", err)
 		return nil
 	}
-	c.client = client
+
+	clientMutex.Lock()
```

Comment:
Just so that I can understand, why the mutex?

Reply:
I was replicating the pre-existing behavior, but I don't really know if that is needed in all the cases. Let me review it, probably is only needed in the case of `Connect` call, that can happen outside the creating process.

Reply:
removed where is not needed.